content/blog/2025/anchor-position-validity.md

```diff
@@ -0,0 +1,284 @@
+  height=200,
+  tab='html,result'
+) }}
+
+There are many reasons that will cause the anchor element will create a
+containing block, and positioned elements that are children will not work.
+
+  - If you set a `position` besides `static` on the anchor and the positioned
+    element is `position: absolute`.
+  - If you transform the element somehow, with `transform`, `translate`,
+    `scale`, etc.
+  - If it's a query container for container size queries.
+
+This is not an exhaustive list. Because there are so many ways to get into this
+situation unexpectedly, I recommend not nesting postioned elements inside the
+anchor.
+
+## Valid anchor pseudo-elements
+
+Most anchors will be elements, but if you're using a pseudo-element, not all
+qualify. The pseudo-element must be a "fully styleable tree-abiding
+pseudo-element". Tree-abiding pseudo-elements behave like regular elements,
+unlike pseudo-elements like `::first-letter` or `::spelling-error`. Some, like
+`::marker` or `::placeholder`, are not fully styleable, as they only allow some
+CSS properties.
+
+The valid pseudo-elements are `::before`, `::after` and
+`::file-selector-button`. `::-webkit-slider-thumb` currently works as an anchor
+in Chrome, but as it is experimental and not part of any CSS spec, it's unclear
+whether it should.
+
+{{ embed.codepen(
+  id='GgKxNWV',
+  title='Valid anchor pseudo-elements',
+  user='jamessw',
+  height=200)
+ }}
+
+## Anchor scope
+
+Anchor scope is great for making reusable anchoring rules, especially if you are
+anchoring on a list item or reusing styles. If you're using `anchor-scope`,
+verify that both the anchor and positioned element are descendants of the
+element with the `anchor-scope` rule, or if the anchor itself has the
+`anchor-scope` rule, that the positioned element is a descendant of the anchor.
+
+## Absolute anchor order
+
+This is the motivation behind the recommended solution to have the anchor come
+before the positioned element in the DOM.
+
+Generally, absolutely positioned elements are rendered after relatively
+positioned elements. If the anchor element is absolutely positioned, then the
+positioned element must come after the anchor in the DOM.
+
+{{ embed.codepen(
+  id='xbKpedP',
+  title='Absolute anchor order',
+  user='jamessw',
+  height=200,
+  tab='html,result')
+ }}
+
+{% callout 'note', false %}
+
+The "after the anchor in the DOM" check happens on the `flat tree`, which means
+that slotted content is placed and shadow hosts are filled with their children.
+
+{% endcallout %}
+
+## Top Layer
+
+If you are using dialogs as modals or popovers, you are creating top layers. If
+the anchor element is in a higher top layer than the positioned element, the
+positioned element will not be able to locate the anchor.
+
+## Anchoring across shadow trees
+
+An element in one tree can anchor to an element in another tree,
+as long as the anchor name is defined in the same tree styles where it is
+referred to.
```

Comment:
I find this wording ("as long as the anchor name is defined in the same tree styles where it is referred to") to be difficult to parse.

Reply:
I elaborated on this- does it make it clearer?